src/appleseed/renderer/kernel/lighting/pt/ptlightingengine.cpp

```diff
@@ -832,20 +840,23 @@ namespace
                 const PhaseFunction* phase_function = medium->get_phase_function();
                 assert(phase_function != nullptr);
 
+                // Any light contribution after a diffuse or glossy bounce is considered indirect.
+                if (ScatteringMode::has_diffuse_or_glossy(vertex.m_scattering_modes))
+                    m_is_indirect_lighting = true;
+
                 Spectrum ray_transmission;
                 phase_function->evaluate_transmission(
                     volume_ray, vertex.m_phase_function_data, ray_transmission);
-                const float density = 1.0f - min_value(ray_transmission);
 
+                const float density = 1.0f - min_value(ray_transmission);
                 const float distance_sample_count_float = density * m_params.m_distance_sample_count;
                 const size_t distance_sample_count = stochastic_cast<size_t>(
                     m_sampling_context, distance_sample_count_float);
```

Comment:
If the number of distance samples is an integer, you no longer need to use `stochastic_cast<>`.